tests/test_schema.py

```diff
@@ -280,13 +281,68 @@ class Model(BaseModel):
                 'title': 'Bulbibulbi!',
                 'default': 'foo',
             },
+            'cara': {'allOf': [{'$ref': '#/definitions/FooBarEnum'}]},
         },
-        'required': ['pikalias'],
+        'required': ['pikalias', 'cara'],
         'title': 'Model',
         'type': 'object',
     }
 
 
+def test_enum_and_model_have_same_behaviour():
+    class Names(str, Enum):
+        rick = 'Rick'
+        morty = 'Morty'
+        summer = 'Summer'
+
+    class Pika(BaseModel):
+        a: str
+
+    class Foo(BaseModel):
+        enum: Names
+        titled_enum: Names = Field(
+            ..., title='Title of enum', description='Description of enum',
+        )
+        model: Pika
+        titled_model: Pika = Field(
+            ..., title='Title of model', description='Description of model',
+        )
+
+    assert Foo.schema() == {
+        'definitions': {
+            'Pika': {
+                'properties': {'a': {'title': 'A', 'type': 'string'}},
+                'required': ['a'],
+                'title': 'Pika',
+                'type': 'object',
+            },
+            'Names': {
+                'description': 'An enumeration.',
+                'enum': ['Rick', 'Morty', 'Summer'],
+                'title': 'Names',
+                'type': 'string',
+            },
+        },
+        'properties': {
+            'enum': {'allOf': [{'$ref': '#/definitions/Names'}]},
+            'model': {'$ref': '#/definitions/Pika'},
```

Comment:
@tiangolo @samuelcolvin I updated the PR as discussed.
Looking at the schema generation for enums and models, I reckon we should (probably in a future PR) try to have the exact same behaviour and use allOf + $ref for models too.

Reply:
Thanks for the attention and double-checking this! :mag: :detective: 

It's because in previous versions of JSON Schema (including the one that the current OpenAPI spec is based on) `$ref` was not allowed to have other keywords alongside of it. So, whenever a `$ref` was used, it had to be the only key in that dictionary/object.

That is allowed in the latest version (I just checked, I actually didn't notice it before). But for compatibility with OpenAPI, which is still based on the previous versions, I think this should work better.

Ref: https://json-schema.org/draft/2019-09/release-notes.html#keyword-changes